lib/logic/player/music_player.dart

```diff
@@ -180,16 +180,22 @@ class _BrowserParentProvider {
   }
 }
 
-class _AudioHandler extends BaseAudioHandler with SeekHandler, WidgetsBindingObserver {
-  _AudioHandler() {
-    _init();
+@visibleForTesting
+class AudioHandler extends BaseAudioHandler with SeekHandler, WidgetsBindingObserver {
+  AudioHandler(MusicPlayer player) {
+    _init(player);
   }
 
   bool _disposed = false;
-  bool _running = false;
-  MusicPlayer player = MusicPlayer.instance;
-
-  Future<void> _init() async {
+  @visibleForTesting
+  bool running = false;
+  late MusicPlayer player;
+  late StreamSubscription playbackSubscriber;
+  late StreamSubscription queueSubscriber;
```

Comment:
This will make calling dispose before init a crash, make those nullable

Reply:
Is calling dispose before init really something that we want to allow? For me crashing there sounds like a good way to find bugs.

Reply:
It should be an OK action to call dispose in not initialized state

Reply:
Actually, `_init` is called in the constructor, it is therefore guaranteed that these fields are initialized.